### DISCUSSION TITLE
Bullet do not set FIXED_CONSTRAINT_TYPE to fixed constraints

### DISCUSSION THREAD

**Discussion by 1vanK** [bulletphysics/bullet3#3565] [General]

**Comment by erwincoumans** [comment]
Please provide a reproduction case.

**Comment by 1vanK** [comment]
src\BulletDynamics\ConstraintSolver\btFixedConstraint.cpp
```
btFixedConstraint::btFixedConstraint(btRigidBody& rbA,btRigidBody& rbB, const btTransform& frameInA,const btTransform& frameInB)
:btGeneric6DofSpring2Constraint(rbA,rbB,frameInA,frameInB)
{
	setAngularLowerLimit(btVector3(0,0,0));
	setAngularUpperLimit(btVector3(0,0,0));
	setLinearLowerLimit(btVector3(0,0,0));
	setLinearUpperLimit(btVector3(0,0,0));
}
```

just set to D6_SPRING_2_CONSTRAINT_TYPE

**Comment by erwincoumans** [comment]
Do you have a suggested fix/patch?

**Comment by 1vanK** [comment]
I wanted to add fixed constraint to Urho3D in the image and likeness of other constraints. They use type in this way
https://github.com/urho3d/Urho3D/blob/master/Source/Urho3D/Physics/Constraint.cpp#L427

But personally, I only recently began to get acquainted with Bullet library so I can not provide any patch

**Comment by erwincoumans** [comment]
You could just use the btGeneric6DofSpring2Constraint directly, and set the linear/angular limits to zero, that would workaround if there is any issue.

Cool, thanks for the report!

**Comment by erwincoumans** [comment]
I moved to pybullet and a C-API, that allows also multibody in an easier way (btMultiBody) instead of btRigidBody and btTypedConstraints. See also the pybullet quickstart guide at http://pybullet.org if you are interested. The underlying C-API is abstracted and very different from the Bullet C++ API.

**Comment by 1vanK** [comment]
Thanks!
